services/server/server.go

```diff
@@ -249,24 +266,44 @@ func (s *Server) ServeGRPC(l net.Listener) error {
 
 // ServeTTRPC provides the containerd ttrpc APIs on the provided listener
 func (s *Server) ServeTTRPC(l net.Listener) error {
+	ctx := context.Background()
+	span := trace.SpanFromContext(ctx)
+	_, ttrpcSpan := (span.Tracer().Start(ctx, "ttrpcSpan"))
+	defer ttrpcSpan.End()
+
 	return trapClosedConnErr(s.ttrpcServer.Serve(context.Background(), l))
 }
 
 // ServeMetrics provides a prometheus endpoint for exposing metrics
 func (s *Server) ServeMetrics(l net.Listener) error {
+	ctx := context.Background()
+	span := trace.SpanFromContext(ctx)
+	_, metricsSpan := (span.Tracer().Start(ctx, "metricsSpan"))
+	defer metricsSpan.End()
+
 	m := http.NewServeMux()
 	m.Handle("/v1/metrics", metrics.Handler())
 	return trapClosedConnErr(http.Serve(l, m))
 }
 
 // ServeTCP allows services to serve over tcp
 func (s *Server) ServeTCP(l net.Listener) error {
+	ctx := context.Background()
+	span := trace.SpanFromContext(ctx)
+	_, tcpSpan := (span.Tracer().Start(ctx, "tcpSpan"))
```

Comment:
Curious is it normal to name a span a span?

Reply:
Not in my experience.  I would probably name these spans after the methods they instrument.